### PR TITLE
Updated hipchat action to support custom colors

### PR DESCRIPTION
**Pull request by matt-digitaliceltd**
As discussed in #1158, have updated hipchat action to support custom colors.

Hope we can merge this in, would be great to not have to reimplement the hipchat action for each project.

### CODE REVIEW

File: lib/fastlane/actions/hipchat.rb [hunk 1]
```diff
@@ -106,6 +110,12 @@ def self.available_options
                                            raise 'No HIPCHAT_API_TOKEN given.'.red
                                          end
                                        end),
+          FastlaneCore::ConfigItem.new(key: :custom_color,
+                                       env_name: "FL_HIPCHAT_CUSTOM_COLOR",
+                                       description: "Specify a custom color, this overrides the success boolean. Can be one of 'yellow', 'red', 'green', 'purple', 'gray', or 'random'",
+                                       optional: true,
+                                       default_value: '',
```
Comment (KrauseFx) [inline lib/fastlane/actions/hipchat.rb]: You can remove the `default_value`

**Comment by KrauseFx** [comment]
Great, thanks for this PR. Could you address the one comment I added and squash all commits into one?

**Comment by digitaliceltd** [comment]
I have made this change and squashed all commits, thanks!

**Comment by KrauseFx** [comment]
That's perfect, thank you so much for your contribution @digitaliceltd :+1:

**Pull request merged**